predict crash location on GCS

EwoudSmeur opened the pull request:
I made a GCS marker to predict the ballistic crash position.

I tried to follow this commit: https://github.com/paparazzi/PprzGCS/commit/b0e1641670fddca0e3a8ce3f43ab100ba6f39a11

I had to take some time to understand the code structure, but probably I still butchered it. Right now there is only one crashItem, which would be a problem if you have multiple drones. I guess it can be done like with DCSHOT with the QList?

EwoudSmeur committed:
predict crash location on GCS

Fabien-B commented:
Since there is a crashItem for each drone, I would put it in the [ACItemManager](https://github.com/paparazzi/PprzGCS/blob/master/src/widgets/map/acitemmanager.h), which hold references to every objects linked to an AC.
It was previously all in the MapWidget, but it quickly became unmanageable, so I created the ACItemManager to organize a little bit. It also made it easier to remove an AC (see the `removeItems` method).

I would also make that optional, as I can imagine it could be a bit disturbing if this waypoint is not desired.
You can use the settings and GlobalConfig : [example](https://github.com/paparazzi/PprzGCS/blob/master/src/app_settings.cpp)

I will try writing a bit of a doc to explain how to use it.

EwoudSmeur commented:
Thanks for the tips @Fabien-B! I'll modify accordingly.

EwoudSmeur committed:
Crashitem as part of AC

EwoudSmeur committed:
checkbox to turn on crash prediction

EwoudSmeur commented:
@Fabien-B I added the prediction as an attribute of an AC and I added a checkbox to enable the display of the crash prediction. The thing I struggled with was to hide the the waypointItem. I now made it very small, but still visible... Maybe you know an easy way to do this?

Fabien-B commented:
I did not anticipated the need to hide a waypoint, as I usually just remove it.
Discussing with Florian, I understand the need of constantly calculating its position, even though it's not displayed, to being able to display it once the drone is crashed in order to search it.
It's probably ok to leave it like that for now, and will think how to hide items on the map.

Fabien-B commented:
I added the virtual method `void setVisible(bool visible)` on MapItem. Can you rebase on master and use it ?

EwoudSmeur committed:
hide crash prediction

EwoudSmeur commented:
Thank you so much @Fabien-B, it works like a charm!

I see your point that it makes sense to compute the crash prediction in the background to be able to display it even after losing the datalink. One could also replay the log and then enable the feature, but it would be less convenient.
